### ISSUE TITLE
Implement Call Caching related REST APIs

### ISSUE THREAD

**Issue by kcibul** [broadinstitute/cromwell#1324]
The following two APIs currently return Not Implemented exceptions.  Need to refine use cases before implementation

POST /workflows/{version}/{id}/call-caching
POST /workflows/{version}/{id}/call-caching/{callFqn}

**Comment by mcovarr** [comment]
Isn't this #1256?

**Comment by katevoss** [comment]
@ktibbett and @jsotobroad does the Green team use this API? Will you need it in order to adopt Cromwell 25?

**Comment by katevoss** [comment]
@mcovarr since you last commented, do these endpoints work yet or no?

**Comment by mcovarr** [comment]
Not only do these not work but they've been removed from Swagger altogether.  I'm not sure what the plan is for these things, maybe @geoffjentry knows.

**Comment by geoffjentry** [comment]
Since I couldn't remember what these endpoints even *did* I had to go [dig into ye olde wayback machine, aka gmail](https://github.com/broadinstitute/cromwell/pull/306/files)

As far as i'm concerned this is along the lines of those "PBE TODO" and similar tickets - I don't remember why we never put this piece of the puzzle back together but clearly it isn't some burning desire. I think I've heard people ask about this functionality once or twice and it gets met with a fuzzy "yeah, i think we used to have that?" but as I said, don't remember our logic.

My $0.02 is to just close this until such a day that people are asking for this again. I realize that's a shockingly hot take considering I just likened it to my stance on other PBE TODO type things :)
